Buildings/ThermalZones/EnergyPlus/Validation/SurfaceComparison/EnergyPlusSurfaces.mo

```diff
@@ -0,0 +1,70 @@
+within Buildings.ThermalZones.EnergyPlus.Validation.SurfaceComparison;
+model EnergyPlusSurfaces
+  "Model that contains Eplus OutputVariable used to validate ZoneSurface"
+  extends Modelica.Icons.Example;
+
+  inner Buildings.ThermalZones.EnergyPlus.Building building(
+    idfName=Modelica.Utilities.Files.loadResource(
+      "modelica://Buildings/Resources/Data/ThermalZones/EnergyPlus/Examples/SingleFamilyHouse_TwoSpeed_ZoneAirBalance/SingleFamilyHouse_TwoSpeed_ZoneAirBalance.idf"),
+    weaName=Modelica.Utilities.Files.loadResource(
+      "modelica://Buildings/Resources/weatherdata/USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.mos"),
+    usePrecompiledFMU=false,
+    showWeatherData=true,
+    computeWetBulbTemperature=false)
+    "Building model"
+    annotation (Placement(transformation(extent={{-80,-60},{-60,-40}})));
+
+  Buildings.ThermalZones.EnergyPlus.OutputVariable TGarAir(
+    name="Zone Mean Air Temperature",
+    key="LIVING ZONE",
+    y(unit="K")) "Garage air temperature"
+    annotation (Placement(transformation(extent={{-80,-20},{-60,0}})));
+  Buildings.ThermalZones.EnergyPlus.OutputVariable TIntWalLivSur(
+    name="Surface Inside Face Temperature",
+    key="Living:Interior",
+    y(unit="K")) "Interior wall living room surface temperature"
+    annotation (Placement(transformation(extent={{-80,20},{-60,40}})));
+  Buildings.ThermalZones.EnergyPlus.OutputVariable TIntWalGarSur(
+    name="Surface Inside Face Temperature",
+    key="Garage:Interior",
+    y(unit="K")) "Interior wall garage surface temperature"
+    annotation (Placement(transformation(extent={{-40,20},{-20,40}})));
+  Buildings.ThermalZones.EnergyPlus.OutputVariable TWesWalGarSur(
+    name="Surface Inside Face Temperature",
+    key="Garage:WestWall",
+    y(unit="K")) "West wall garage surface temperature"
+    annotation (Placement(transformation(extent={{-40,60},{-20,80}})));
+  Buildings.ThermalZones.EnergyPlus.OutputVariable TEasWalGarSur(
+    name="Surface Inside Face Temperature",
+    key="Garage:EastWall",
+    y(unit="K")) "East wall garage surface temperature"
+    annotation (Placement(transformation(extent={{-80,60},{-60,80}})));
+
+  annotation (
+    Documentation(
+      info="<html>
+<p>
+This model simulates the envelope of a single family house with EnergyPlus and is used
```

Comment:
This test outputs 5 temperatures at the top level, the plot shows 4 trajectories, 3 of which match each other quite well. Are you plotting the right 4 quantities, and why not plotting the 5th temperature? It would be good to describe what the user should expect from the result, e.g., which trajectories should be close to each other?

Maybe you could use an assertion to stop the simulation if the difference is larger than what you think is acceptable.

Reply:
This tests looks at the air temperature of the garage and the surface temperature of two of its exterior walls and the interior wall towards the living room. There is one additional output being read from EnergyPlus and that is the surface temperature of the same interior wall but on the side of the living room. The garage was chosen since it is an unconditonned, free-floating zone. The living room side of the interior wall is read to be used as a boundary condition to the interior wall heat transfer in the `ZoneSurface` case, but it is a bit redundant to validate its value, since an issue with the interior wall would be reflected in the other side temperature.

The main objective of this test is to compare the value obtained with and without the spawn object `ZoneSurface` to replace the heat transfer calculation through the exterior and interior walls. So one test operates as the baseline (`EnergyPlusSurfaces`) and the other is the spawn surface object validation (`ZoneSurfaces`).

I've updated the *.mos so that both file are simulated at the same time and actually compared in the same plot. I've also moved the `EnergyPlusSurfaces` example model as a base class, renamed it `ReferenceSurfaces` and removed its *mos command. Does that structure make sense? Note that BuildingsPy does not save the reference results for the second `mat` file generated, but it's not a problem in this case since the second `mat` is the reference results.